src/main/java/com/flowpowered/nbt/regionfile/RegionFile.java

```diff
@@ -0,0 +1,235 @@
+package com.flowpowered.nbt.regionfile;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.IntBuffer;
+import java.nio.ShortBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.BitSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * This helper class provides functionality to read the data of single chunks in a region/anvil file. It uses modern {@code java.nio}
+ * classes like {@link Path} and {@link FileChannel} to access its data from the file. Each instance of the class represents a single file,
+ * which will be loaded completely in the constructor. Warning: typical Minecraft region files may take up to several MiB of space and the
+ * theoretical maximum size is at one GiB (1024 region files take up at most 255 sectors à 4096 bytes). <br/>
+ * Once the object has been created, it will close the opened {@link FileChannel}. No data will be read anymore. It will be assumed that the
+ * file will not be modified externally in any ways. All changes to the objects are kept in memory and only written back to the file upon
+ * calling {@link #writeChanges()}, in which case a new file channel is opened.
+ *
+ * @author piegames
+ */
+public class RegionFile implements Iterable<Chunk> {
+
+	protected final Path	file;
+
+	protected final Chunk[]	chunks;
```

Comment:
Don't cache chunks! getChunk() and setChunk() should directly load and save chunks.